src/shop.rs

```diff
@@ -26,7 +26,23 @@ impl Shop {
     /// this store wins, FightResult::Loss if this store loses, and a
     /// FightResult::Tie if both stores win the same number of battles.
     pub fn fight_store(&self, other: &Shop) -> FightResult {
-        todo!()
+        let mut tally = 0;
+        for card in self.cards.iter() {
+            for other_card in other.cards.iter() {
+                let result = card.print_fight(other_card);
+                match result {
+                    FightResult::Win => tally += 1,
+                    FightResult::Loss => tally -= 1,
+                    _ => (),
+                }
+            }
+        }
```

Comment:
It's a bit funky, but here's another way to solve this that just uses iterators:

```rust
let tally = self
    .cards
    .iter()
    .flat_map(|card| {
        other.cards.iter().map(move |other_card| {
            let result = card.print_fight(other_card);
            match result {
                FightResult::Win => 1,
                FightResult::Loss => -1,
                _ => 0,
            }
        })
    })
    .sum();
```

Where we iterate over the first store's cards, then over the second, and collect `+1` or `-1` for each battle, then add them up at the end. We use `flat_map` so that even though we get something that might look like `[1,-1,-1], [0, 1, -1]...` for each of the cards in the first store, it "flattens" it to just `[1, -1, -1, 0, 1, -1]`, then we can just `sum()` that together :)

Though I don't think I got far enough in the fall to talk about iterator methods, so this is mostly just extra info /shrug